src/radix_connect/well_known_client/client.rs

```diff
@@ -0,0 +1,116 @@
+use crate::prelude::*;
+
+const SUFFIX_WELL_KNOWN_FILE: &str = ".well-known/dapp.json";
+
+pub struct WellKnownClient {
+    pub http_client: HttpClient,
+}
+
+impl WellKnownClient {
+    pub fn new(http_client: HttpClient) -> Self {
+        Self { http_client }
+    }
+
+    pub fn new_with_network_antenna(
+        network_antenna: Arc<dyn NetworkAntenna>,
+    ) -> Self {
+        Self::new(HttpClient::new(network_antenna))
+    }
+}
+
+impl NetworkRequest {
+    fn get_well_known(url: Url) -> Self {
+        let well_known_url = url.join(SUFFIX_WELL_KNOWN_FILE).unwrap();
+        NetworkRequest::new_get(well_known_url)
+    }
+}
+
+/// The Well Known Client .
+/// It will be used to fetch the well-known file for a given origin.
+impl WellKnownClient {
+    /// Fetches the well-known file, internally it appends the suffix to the origin URL.
+    pub async fn get_well_known_file(
+        &self,
+        origin: Url,
+    ) -> Result<DappDefinitions> {
+        let network_request = NetworkRequest::get_well_known(origin);
+        self.http_client
+            .execute_request_with_decoding(network_request)
+            .await
+    }
+}
+
+#[cfg(test)]
+impl WellKnownClient {
+    fn new_always_failing() -> Self {
+        Self::new_with_network_antenna(Arc::new(
+            MockAntenna::new_always_failing(),
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use actix_rt::time::timeout;
+    use std::time::Duration;
+    const MAX: Duration = Duration::from_millis(10);
+
+    const TEST_ORIGIN: &str = "https://d2xmq49o1iddud.cloudfront.net";
+
+    #[test]
+    fn test_suffix_is_correct() {
+        assert_eq!(SUFFIX_WELL_KNOWN_FILE, ".well-known/dapp.json")
+    }
+
+    #[test]
+    fn test_get_well_known_request_error() {
+        // ARRANGE
+        let url = Url::parse(TEST_ORIGIN).unwrap();
+        // ACT
+        let request = NetworkRequest::get_well_known(url.clone());
+        // ASSERT
+        assert_eq!(request.url, url.join(SUFFIX_WELL_KNOWN_FILE).unwrap());
+        assert_eq!(request.method, NetworkMethod::Get);
+    }
+
+    #[actix_rt::test]
+    async fn test_get_well_known_file_correct_request_made() {
+        // ARRANGE
```

Comment:
nice! just we a few `ARRANGE`, `ACT`, `ASSERT` this tests became much more clear! especially good in this case when the `ASSERT` is spelled in lines above the `ACT`, it helps me a lot, and others too I'm sure!